BR_Doris1115.sql

```diff
@@ -0,0 +1,66 @@
+USE Proj_B1
+
+/*
+4. Enforce the business rule to prevent new inserts into EMP_SHIFT_STATUS for a shift that has already passed
+ */
+CREATE FUNCTION FN_noPassedESS()
+RETURNS INT
+AS
+BEGIN
+    DECLARE @RET INT = 0
+    IF EXISTS (
+        SELECT * FROM tblEMP_SHIFT_STATUS ESS
+            JOIN tblSTATUS St ON St.StatusID = ESS.StatusID
+            JOIN tblSHIFT S ON S.ShiftID = ESS.ShiftID
+            JOIN tblMONTH M ON M.MonthID = S.MonthID
+            JOIN tblDay D ON D.DayID = S.DayID
+            JOIN tblHOUR H ON H.HourID = S.BeginHourID
+        WHERE (St.StatusTitle = 'Worked') OR
```

Comment:
Is this necessary?

Reply:
What if there is a shift that has not been worked but the time passed? Is it possible?